src/components/contracts/ContractsDialog.js

```diff
@@ -102,6 +104,10 @@ const ContractsDialog = ({
   };
   const handleSave = () => {
     dispatch(setIsLoading(true));
+
+    if (currentContract.fieldValues.contract_main_orgunit) {
+      currentContract.fieldValues.contract_main_orgunit = currentContract.fieldValues.contract_main_orgunit.id;
```

Comment:
currentContract is part of the state, use setCurrentContract to change his value